src/Functions/array/randomSampleFromArray.cpp

```diff
@@ -0,0 +1,126 @@
+/// arrayRandomSample(arr, k) - Returns k random elements from the input array
+class FunctionArrayRandomSample : public IFunction
+{
+public:
+    static constexpr auto name = "arrayRandomSample";
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionArrayRandomSample>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 2; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        const DataTypeArray * array_type = checkAndGetDataType<DataTypeArray>(arguments[0].get());
+        if (!array_type)
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "First argument must be an array");
+
+        if (!isUnsignedInteger(arguments[1]))
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument must be a unsigned integer");
+
+        // Return an array with the same nested type as the input array
+        return std::make_shared<DataTypeArray>(array_type->getNestedType());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        const ColumnArray * column_array = checkAndGetColumn<ColumnArray>(arguments[0].column.get());
+        if (!column_array)
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "First argument must be an array");
+
+        const IColumn * col_num = arguments[1].column.get();
+
+        if (!col_num || col_num->empty())
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "The second argument column is empty or null, type = {}",
+                arguments[1].type->getName());
+        }
+
+        UInt64 K;
+        try
+        {
+            K = col_num->getUInt(0);
+        }
+        catch (...)
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Failed to fetch UInt64 from the second argument column, type = {}",
+                arguments[1].type->getName());
+        }
+
+        Poco::Logger::get("FunctionRandomSampleFromArray").debug("The number of samples K = " + std::to_string(K));
+
+        const auto & offsets = column_array->getOffsets();
+        size_t num_elements = offsets[0];
+
+        Poco::Logger::get("FunctionRandomSampleFromArray").debug("The number of elements in the array = " + std::to_string(num_elements));
+
+        if (num_elements == 0 || K == 0)
+        {
+            // Handle edge cases where input array is empty or K is 0
+            return column_array->cloneEmpty();
+        }
+
+        std::random_device rd;
+        std::mt19937 gen(rd());
+
+        if (static_cast<UInt64>(num_elements) < K)
```

Comment:
Should be sth like: K = std::min(K, num_elements)

Reply:
Yep, it is also acceptable :slightly_smiling_face:

Reply:
done